manual_flagging/src/flagging_rolling.py

```diff
@@ -1079,27 +1088,25 @@ def split_logic(words: str):
 
 def name_selector(tokens) -> str:
     """
-    Attempts to select the last name of a persons name based on number of tokens.
+    Attempts to select the last name of a person's name based on the number of tokens.
     Inputs:
-        tokens: name to be identified
+        tokens: list of strings where each string is a name token
     Outputs:
         'Empty Name' if name is empty.
         id (str): identified last name
     """
-    if tokens == "Empty Name":
-        return tokens
-    # Ex: John Smith Jr
-    if tokens[-1] in ["jr", "sr", "ii", "iii", "iv", "v"]:
+
+    suffixes = ["jr", "sr", "ii", "iii", "iv", "v"]
+
+    if tokens == "Empty Name" or tokens == []:
+        return "Empty Name"
+
+    while tokens[-1] in suffixes:
         tokens = tokens[:-1]
-    # Ex: John Smith
-    if len(tokens) == 2:
-        id = tokens[1]
-    # John George Smith
-    if len(tokens) == 3:
-        id = tokens[2]
-    # John George Theodore Smith
-    else:
-        id = tokens[-1]
+        if not tokens:  # Avoids IndexError if all tokens are removed.
+            return "Empty Name"
```

Comment:
This fixes the problem of multiple suffixes like `'Bob jr iii'`

Reply:
I appreciate the handling for the unfortunate soul named `Jr. Jr. Jr.`.